### PR TITLE
Rm gunclass

### PR DESCRIPTION
**Pull request by rabiyamajeed**

### CODE REVIEW

**Comment by rabiyamajeed** [comment]
gun class

File: gun.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	Gun::durability = 0;
+	Gun::recoil = 0;
+}
+
+Gun::~Gun() {}
+
+int Gun::get_ammo() { return Gun::ammo; }
+int Gun::get_durability() { return Gun::durability; }
+int Gun::get_length() { return Gun::length; }
+int Gun::get_width() { return Gun::width; }
```
Comment (zzakiullah) [inline gun.cpp]: What's the difference between length and width? If it's referring to the dimensions of the gun, the object class already has width and height property so the gun class will inherit from it, so this isn't needed. (unless this is referring to something else)

**Review by zzakiullah — CHANGES_REQUESTED** [review]

File: gun.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#pragma once
+// Object.cpp
```
Comment (zzakiullah) [inline gun.h]: Also should be gun.h lol

File: gun.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Object.cpp
```
Comment (zzakiullah) [inline gun.cpp]: Not really important but should be gun.cpp lol

**Review by zzakiullah — CHANGES_REQUESTED** [review]

File: gun.cpp (hunk 1 above)
Comment (rabiyamajeed) [reply gun.cpp]: okai! this was just gun dimensions so i will take out

File: Gun.cpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+Gun::~Gun() {}
+
+int Gun::get_ammo() { return Gun::ammo; }
+int Gun::get_durability() { return Gun::durability; }
+int Gun::get_shooting_speed() { return Gun::shooting_speed; }
+double Gun::get_accuracy() { return Gun::accuracy; }
+int Gun::get_durability() { return Gun::durability; }
+double Gun::get_recoil() { return Gun::recoil; }
+
+void Gun::set_ammo(int new_ammo) { Gun::ammo = ammo; }
```
Comment (zzakiullah) [inline Gun.cpp]: You have the parameter as "new_ammo" but you're saying Gun::ammo = ammo, which just resets it to itself. I think you should change the parameter to just "ammo" in the .cpp and .h file (same for the other set functions)

File: Gun.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	int get_ammo();
+	int get_durability();
+	int get_shooting_speed();
+	double get_accuracy();
+	int get_durability();
+	double get_recoil();
+
+
+
+	void set_ammo(int new_ammo);
```
Comment (zzakiullah) [inline Gun.h]: Change this to "ammo" (take out the new_ in the other set functions too)

**Review by zzakiullah — CHANGES_REQUESTED** [review]
Check setting functions

File: Gun.cpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+int Gun::get_durability() { return Gun::durability; }
+int Gun::get_shooting_speed() { return Gun::shooting_speed; }
+double Gun::get_accuracy() { return Gun::accuracy; }
+int Gun::get_durability() { return Gun::durability; }
+double Gun::get_recoil() { return Gun::recoil; }
+
+void Gun::set_ammo(int ammo) { Gun::ammo = ammo; }
+void Gun::set_durability(int durability) { Gun::durability = durability; }
+void Gun::set_shooting_speed(int shooting_speed) { Gun::shooting_speed = shooting_speed; }
+void Gun::set_accuracy(int accuracy) { Gun::accuracy = accuracy; }
```
Comment (zzakiullah) [inline Gun.cpp]: accuracy is a double

File: Gun.cpp [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+void Gun::set_durability(int durability) { Gun::durability = durability; }
+void Gun::set_recoil(int recoil) { Gun::recoil = recoil; }
```
Comment (zzakiullah) [inline Gun.cpp]: recoil is a double

File: Gun.h [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+	double get_accuracy();
+	int get_durability();
+	double get_recoil();
+
+
+
+	void set_ammo(int ammo);
+	void set_durability(int durability);
+	void set_shooting_speed(int shoothing_speed);
+	void set_accuracy(int accuracy);
```
Comment (zzakiullah) [inline Gun.h]: accuracy is a double

File: Gun.h [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+	void set_durability(int durability);
+	void set_recoil(int recoil);
```
Comment (zzakiullah) [inline Gun.h]: recoil is a double

**Review by zzakiullah — CHANGES_REQUESTED** [review]

File: Gun.h [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+	int get_durability();
+	int get_shooting_speed();
+	double get_accuracy();
+	double get_recoil();
+
+
+
+	void set_ammo(int ammo);
+	void set_durability(int durability);
+	void set_shooting_speed(int shoothing_speed);
```
Comment (zzakiullah) [inline Gun.h]: shooting_speed

**Review by zzakiullah — CHANGES_REQUESTED** [review]

**Review by zzakiullah — APPROVED** [review]
Coolio I will forever laugh at shoothing

**Review by s-yfang — APPROVED** [review]

**Pull request merged**